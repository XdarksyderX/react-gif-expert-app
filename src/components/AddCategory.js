import React, { useState } from 'react';
import PropTypes from 'prop-types';
export const AddCategory = ({ setCategories }) => {
	const [inputValue, setInputValue] = useState('');

	const handleInputChange = ({ target }) => {
		setInputValue(target.value);
	};

	const handleSubmit = (e) => {
		e.preventDefault();
		if (inputValue.trim().length > 2) {
			setCategories((prev) => [inputValue, ...prev]);
			setInputValue('');
		}
	};

	return (
		<form onSubmit={handleSubmit}>
			<input type="text" value={inputValue} onChange={handleInputChange} />
		</form>
	);
};

AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired,
};
