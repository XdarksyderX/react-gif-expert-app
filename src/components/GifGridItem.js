import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, url }) => {
	return (
		<div className="card animate__animated animate__fadeIn animate__delay-1s">
			<img src={url} alt={title} />
			<p> {title} </p>
		</div>
	);
};

GifGridItem.propTypes = {
	url: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired
}