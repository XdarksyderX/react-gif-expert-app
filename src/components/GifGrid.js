import React from 'react';
import PropTypes from 'prop-types';

import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {
	const { data: images, loading } = useFetchGifs(category);
	return (
		<div className="gif-grid animate__animated animate__bounceInDown">
			<h3>{category}</h3>
			{loading ? <p className="animate__animated animate__fadeIn animate__infinite">Loading...</p> : null}
			<ul className="gifs-container">
				{images.map((img) => {
					return <GifGridItem key={img.id} url={img.url} title={img.title} />;
				})}
			</ul>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};
