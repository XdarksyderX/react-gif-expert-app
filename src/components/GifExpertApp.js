import React, { useState } from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['One Punch']);

	return (
		<div className="app">
			<h2>GifExpertApp</h2>
			
			<AddCategory setCategories={setCategories}/>
            <ul>
                {categories.map(cat => <GifGrid key={cat} category={cat}/>)}
            </ul>
			<hr />
		</div>
	);
};
