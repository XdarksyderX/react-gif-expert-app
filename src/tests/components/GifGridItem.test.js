import React from 'react';
import { shallow } from 'enzyme';

import { GifGridItem } from '../../components/GifGridItem';

describe('GifGrid', () => {
	const url = 'asasdd';
	const title = 'asdasd';

	let wrapper = shallow(<GifGridItem url={url} title={title} />);

	test('Show correctly the component', () => {
        wrapper = shallow(<GifGridItem url={url} title={title} />);
		expect(wrapper).toMatchSnapshot();
	});

    test('Should a paragraph with the title', () => {
        wrapper = shallow(<GifGridItem url={url} title={title} />);
        const paragraph = wrapper.find('p');
        expect(paragraph.text().trim()).toBe(title);
    });

    test('Image`s props in correct place ', () => {
        wrapper = shallow(<GifGridItem url={url} title={title} />);
        const img = wrapper.find('img');
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })

    test('should have "animate__bounceInDown" animation', () => {
        wrapper = shallow(<GifGridItem url={url} title={title} />);
        const div = wrapper.find('div');
        const classes = div.prop('className');
        

        expect(classes.includes('animate__fadeIn')).toBe(true);
    })
    
    
    
});
