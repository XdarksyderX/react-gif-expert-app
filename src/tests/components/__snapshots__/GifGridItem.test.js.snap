// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifGrid Show correctly the component 1`] = `ShallowWrapper {}`;
